File: web/GuestHouseTry5/src/AppBundle/Entity/House.php
<?php

namespace AppBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * House
 *
 * @ORM\Table(name="house", uniqueConstraints={@ORM\UniqueConstraint(name="id", columns={"id"})}, indexes={@ORM\Index(name="fk_user", columns={"fk_user"})})
 * @ORM\Entity
 */
class House
{
    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="country", type="string", length=30, nullable=false)
     */
    private $country;

    /**
     * @var string
     *
     * @ORM\Column(name="town", type="string", length=30, nullable=false)
     */
    private $town;

    /**
     * @var string
     *
     * @ORM\Column(name="nature", type="string", length=30, nullable=false)
     */
    private $nature;

    /**
     * @var integer
     *
     * @ORM\Column(name="price", type="integer", nullable=false)
     */
    private $price;

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="text", length=65535, nullable=true)
     */
    private $description;

    /**
     * @var integer
     *
     * @ORM\Column(name="rating", type="integer", nullable=true)
     */
    private $rating;

    /**
     * @var boolean
     *
     * @ORM\Column(name="airconditioner", type="boolean", nullable=true)
     */
    private $airconditioner;

    /**
     * @var boolean
     *
     * @ORM\Column(name="fireplace", type="boolean", nullable=true)
     */
    private $fireplace;

    /**
     * @var boolean
     *
     * @ORM\Column(name="washingmachine", type="boolean", nullable=true)
     */
    private $washingmachine;

    /**
     * @var boolean
     *
     * @ORM\Column(name="dishwasher", type="boolean", nullable=true)
     */
    private $dishwasher;

    /**
     * @var boolean
     *
     * @ORM\Column(name="dvd", type="boolean", nullable=true)
     */
    private $dvd;

    /**
     * @var boolean
     *
     * @ORM\Column(name="indoorpool", type="boolean", nullable=true)
     */
    private $indoorpool;

    /**
     * @var boolean
     *
     * @ORM\Column(name="sauna", type="boolean", nullable=true)
     */
    private $sauna;

    /**
     * @var boolean
     *
     * @ORM\Column(name="jacuzzi", type="boolean", nullable=true)
     */
    private $jacuzzi;

    /**
     * @var boolean
     *
     * @ORM\Column(name="tv", type="boolean", nullable=true)
     */
    private $tv;

    /**
     * @var boolean
     *
     * @ORM\Column(name="elevator", type="boolean", nullable=true)
     */
    private $elevator;

    /**
     * @var boolean
     *
     * @ORM\Column(name="barbecue", type="boolean", nullable=true)
     */
    private $barbecue;

    /**
     * @var boolean
     *
     * @ORM\Column(name="garden", type="boolean", nullable=true)
     */
    private $garden;

    /**
     * @var boolean
     *
     * @ORM\Column(name="childrengames", type="boolean", nullable=true)
     */
    private $childrengames;

    /**
     * @var boolean
     *
     * @ORM\Column(name="outdoorparking", type="boolean", nullable=true)
     */
    private $outdoorparking;

    /**
     * @var boolean
     *
     * @ORM\Column(name="indoorparking", type="boolean", nullable=true)
     */
    private $indoorparking;

    /**
     * @var boolean
     *
     * @ORM\Column(name="gardenfurniture", type="boolean", nullable=true)
     */
    private $gardenfurniture;

    /**
     * @var boolean
     *
     * @ORM\Column(name="terrace", type="boolean", nullable=true)
     */
    private $terrace;

    /**
     * @var string
     *
     * @ORM\Column(name="other", type="string", length=50, nullable=true)
     */
    private $other;

    /**
     * @var string
     *
     * @ORM\Column(name="picture1", type="blob", nullable=true)
     */
    private $picture1;

    /**
     * @var string
     *
     * @ORM\Column(name="picture2", type="blob", nullable=true)
     */
    private $picture2;

    /**
     * @var string
     *
     * @ORM\Column(name="picture3", type="blob", nullable=true)
     */
    private $picture3;

    /**
     * @var \User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="fk_user", referencedColumnName="id")
     * })
     */
    private $fkUser;

    /**
     * @var \Doctrine\Common\Collections\Collection
     *
     * @ORM\ManyToMany(targetEntity="User", inversedBy="fkStayHouseid")
     * @ORM\JoinTable(name="stay",
     *   joinColumns={
     *     @ORM\JoinColumn(name="fk_stay_houseid", referencedColumnName="id")
     *   },
     *   inverseJoinColumns={
     *     @ORM\JoinColumn(name="fk_stay_userid", referencedColumnName="id")
     *   }
     * )
     */
    private $fkStayUserid;

    /**
     * Constructor
     */
    public function __construct()
    {
        $this->fkStayUserid = new \Doctrine\Common\Collections\ArrayCollection();
    }

    /**
     * @return int
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @param int $id
     */
    public function setId($id)
    {
        $this->id = $id;
    }

    /**
     * @return string
     */
    public function getCountry()
    {
        return $this->country;
    }

    /**
     * @param string $country
     */
    public function setCountry($country)
    {
        $this->country = $country;
    }

    /**
     * @return string
     */
    public function getTown()
    {
        return $this->town;
    }

    /**
     * @param string $town
     */
    public function setTown($town)
    {
        $this->town = $town;
    }

    /**
     * @return string
     */
    public function getNature()
    {
        return $this->nature;
    }

    /**
     * @param string $nature
     */
    public function setNature($nature)
    {
        $this->nature = $nature;
    }

    /**
     * @return int
     */
    public function getPrice()
    {
        return $this->price;
    }

    /**
     * @param int $price
     */
    public function setPrice($price)
    {
        $this->price = $price;
    }

    /**
     * @return string
     */
    public function getDescription()
    {
        return $this->description;
    }

    /**
     * @param string $description
     */
    public function setDescription($description)
    {
        $this->description = $description;
    }

    /**
     * @return int
     */
    public function getRating()
    {
        return $this->rating;
    }

    /**
     * @param int $rating
     */
    public function setRating($rating)
    {
        $this->rating = $rating;
    }

    /**
     * @return boolean
     */
    public function isAirconditioner()
    {
        return $this->airconditioner;
    }

    /**
     * @param boolean $airconditioner
     */
    public function setAirconditioner($airconditioner)
    {
        $this->airconditioner = $airconditioner;
    }

    /**
     * @return boolean
     */
    public function isFireplace()
    {
        return $this->fireplace;
    }

    /**
     * @param boolean $fireplace
     */
    public function setFireplace($fireplace)
    {
        $this->fireplace = $fireplace;
    }

    /**
     * @return boolean
     */
    public function isWashingmachine()
    {
        return $this->washingmachine;
    }

    /**
     * @param boolean $washingmachine
     */
    public function setWashingmachine($washingmachine)
    {
        $this->washingmachine = $washingmachine;
    }

    /**
     * @return boolean
     */
    public function isDishwasher()
    {
        return $this->dishwasher;
    }

    /**
     * @param boolean $dishwasher
     */
    public function setDishwasher($dishwasher)
    {
        $this->dishwasher = $dishwasher;
    }

    /**
     * @return boolean
     */
    public function isDvd()
    {
        return $this->dvd;
    }

    /**
     * @param boolean $dvd
     */
    public function setDvd($dvd)
    {
        $this->dvd = $dvd;
    }

    /**
     * @return boolean
     */
    public function isIndoorpool()
    {
        return $this->indoorpool;
    }

    /**
     * @param boolean $indoorpool
     */
    public function setIndoorpool($indoorpool)
    {
        $this->indoorpool = $indoorpool;
    }

    /**
     * @return boolean
     */
    public function isSauna()
    {
        return $this->sauna;
    }

    /**
     * @param boolean $sauna
     */
    public function setSauna($sauna)
    {
        $this->sauna = $sauna;
    }

    /**
     * @return boolean
     */
    public function isJacuzzi()
    {
        return $this->jacuzzi;
    }

    /**
     * @param boolean $jacuzzi
     */
    public function setJacuzzi($jacuzzi)
    {
        $this->jacuzzi = $jacuzzi;
    }

    /**
     * @return boolean
     */
    public function isTv()
    {
        return $this->tv;
    }

    /**
     * @param boolean $tv
     */
    public function setTv($tv)
    {
        $this->tv = $tv;
    }

    /**
     * @return boolean
     */
    public function isElevator()
    {
        return $this->elevator;
    }

    /**
     * @param boolean $elevator
     */
    public function setElevator($elevator)
    {
        $this->elevator = $elevator;
    }

    /**
     * @return boolean
     */
    public function isBarbecue()
    {
        return $this->barbecue;
    }

    /**
     * @param boolean $barbecue
     */
    public function setBarbecue($barbecue)
    {
        $this->barbecue = $barbecue;
    }

    /**
     * @return boolean
     */
    public function isGarden()
    {
        return $this->garden;
    }

    /**
     * @param boolean $garden
     */
    public function setGarden($garden)
    {
        $this->garden = $garden;
    }

    /**
     * @return boolean
     */
    public function isChildrengames()
    {
        return $this->childrengames;
    }

    /**
     * @param boolean $childrengames
     */
    public function setChildrengames($childrengames)
    {
        $this->childrengames = $childrengames;
    }

    /**
     * @return boolean
     */
    public function isOutdoorparking()
    {
        return $this->outdoorparking;
    }

    /**
     * @param boolean $outdoorparking
     */
    public function setOutdoorparking($outdoorparking)
    {
        $this->outdoorparking = $outdoorparking;
    }

    /**
     * @return boolean
     */
    public function isIndoorparking()
    {
        return $this->indoorparking;
    }

    /**
     * @param boolean $indoorparking
     */
    public function setIndoorparking($indoorparking)
    {
        $this->indoorparking = $indoorparking;
    }

    /**
     * @return boolean
     */
    public function isGardenfurniture()
    {
        return $this->gardenfurniture;
    }

    /**
     * @param boolean $gardenfurniture
     */
    public function setGardenfurniture($gardenfurniture)
    {
        $this->gardenfurniture = $gardenfurniture;
    }

    /**
     * @return boolean
     */
    public function isTerrace()
    {
        return $this->terrace;
    }

    /**
     * @param boolean $terrace
     */
    public function setTerrace($terrace)
    {
        $this->terrace = $terrace;
    }

    /**
     * @return string
     */
    public function getOther()
    {
        return $this->other;
    }

    /**
     * @param string $other
     */
    public function setOther($other)
    {
        $this->other = $other;
    }

    /**
     * @return string
     */
    public function getPicture1()
    {
        return $this->picture1;
    }

    /**
     * @param string $picture1
     */
    public function setPicture1($picture1)
    {
        $this->picture1 = $picture1;
    }

    /**
     * @return string
     */
    public function getPicture2()
    {
        return $this->picture2;
    }

    /**
     * @param string $picture2
     */
    public function setPicture2($picture2)
    {
        $this->picture2 = $picture2;
    }

    /**
     * @return string
     */
    public function getPicture3()
    {
        return $this->picture3;
    }

    /**
     * @param string $picture3
     */
    public function setPicture3($picture3)
    {
        $this->picture3 = $picture3;
    }

    /**
     * @return \User
     */
    public function getFkUser()
    {
        return $this->fkUser;
    }

    /**
     * @param \User $fkUser
     */
    public function setFkUser($fkUser)
    {
        $this->fkUser = $fkUser;
    }

    /**
     * @return \Doctrine\Common\Collections\Collection
     */
    public function getFkStayUserid()
    {
        return $this->fkStayUserid;
    }

    /**
     * @param \Doctrine\Common\Collections\Collection $fkStayUserid
     */
    public function setFkStayUserid($fkStayUserid)
    {
        $this->fkStayUserid = $fkStayUserid;
    }



}

File: desktop/GuestHousetry3/src/entities/House.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

import entitiesDao.CommentDAO;
import entitiesDao.RateDAO;
import java.awt.Image;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 *
 * @author grami
 */
public class House {
    
   protected int id;
   protected String country;
   protected String town;
   protected String nature;
   protected int price;
   protected String description;
   protected int rating;
   protected boolean airconditioner;
   protected boolean fireplace;
   protected boolean washingmachine;
   protected boolean dishwasher;
   protected boolean dvd;
   protected boolean indoorpool;
   protected boolean sauna;
   protected boolean jacuzzi;
   protected boolean tv;
   protected boolean elevator;
   protected boolean barbecue;
   protected boolean garden;
   protected boolean childrengames;
   protected boolean outdoorparking;
   protected boolean indoorparking;
   protected boolean gardenfurniture;
   protected boolean terrace;
   protected String other;
   protected int equipment =0;
   protected Image picture1;
   protected Image picture2;
   protected Image picture3;

    public int getEquipment() {
        return equipment;
    }

    public void setEquipment() {
        if(isAirconditioner())
            equipment++;
        if(isBarbecue())
            equipment++;
        if(isChildrengames())
            equipment++;
        if(isDishwasher())
            equipment++;
        if(isDvd())
            equipment++;
        if(isElevator())
            equipment++;
        if(isFireplace())
            equipment++;
        if(isGarden())
            equipment++;
        if(isGardenfurniture())
            equipment++;
        if(isIndoorparking())
            equipment++;
        if(isIndoorpool())
            equipment++;
        if(isJacuzzi())
            equipment++;
        if(isOutdoorparking())
            equipment++;
        if(isSauna())
            equipment++;
        if(isTerrace())
            equipment++;
        if(isTv())
            equipment++;
        if(isWashingmachine())
            equipment++;
        if(other!=null)
            equipment++;
    }
   
   protected List<Rate> rates;
   protected Map<Comment,User> comments;
   
   public void addRates(){
       
       RateDAO rd=new RateDAO();
       if(rates==null)
           rates=new ArrayList<>();
       rates=rd.findbyHouseID(id);   
}
   
   public void addComments(){
       CommentDAO rd=new CommentDAO();
       if(comments==null)
           comments=new HashMap<>();
       comments=rd.findbyHouseID(id);       
   }
    public House() {
        
    }

    public List<Rate> getRates() {
        return rates;
    }

    public Map<Comment, User> getComments() {
        return comments;
    }

    public House(String country, String town, String nature, int price, String description) {
        this.country = country;
        this.town = town;
        this.nature = nature;
        this.price = price;
        this.description = description;
        setEquipment();
    }

    public House(String country, String town, String nature, int price) {
        this.country = country;
        this.town = town;
        this.nature = nature;
        this.price = price;
        setEquipment();
    }
        
    public House(House h){
        this(h.id, h.country, h.town, h.nature, h.price, h.description, h.rating,
                h.airconditioner, h.fireplace, h.washingmachine, h.dishwasher, h.dvd,
                h.indoorpool, h.sauna, h.jacuzzi, h.tv, h.elevator, h.barbecue, h.garden,
                h.childrengames, h.outdoorparking, h.indoorparking, h.gardenfurniture, h.terrace, h.other);
        setEquipment();
    }
    
    public House(int id, String country, String town, String nature, int price, int rating) {
        this.id = id;
        this.country = country;
        this.town = town;
        this.nature = nature;
        this.price = price;
        this.rating = rating;
        setEquipment();
    }

    public Image getPicture1() {
        return picture1;
    }

    public void setPicture1(Image picture1) {
        this.picture1 = picture1;

    }

    public Image getPicture2() {
        return picture2;
    }

    public void setPicture2(Image picture2) {
        this.picture2 = picture2;
    }

    public Image getPicture3() {
        return picture3;
    }

    public void setPicture3(Image picture3) {
        this.picture3 = picture3;
    }

    public House(int id, String country, String town, String nature, int price, String description, int rating, boolean airconditioner, boolean fireplace, boolean washingmachine, boolean dishwasher, boolean dvd, boolean indoorpool, boolean sauna, boolean jacuzzi, boolean tv, boolean elevator, boolean barbecue, boolean garden, boolean childrengames, boolean outdoorparking, boolean indoorparking, boolean gardenfurniture, boolean terrace, String other, Image picture1, Image picture2,Image picture3) {
        this.id = id;
        this.country = country;
        this.town = town;
        this.nature = nature;
        this.price = price;
        this.description = description;
        this.rating = rating;
        this.airconditioner = airconditioner;
        this.fireplace = fireplace;
        this.washingmachine = washingmachine;
        this.dishwasher = dishwasher;
        this.dvd = dvd;
        this.indoorpool = indoorpool;
        this.sauna = sauna;
        this.jacuzzi = jacuzzi;
        this.tv = tv;
        this.elevator = elevator;
        this.barbecue = barbecue;
        this.garden = garden;
        this.childrengames = childrengames;
        this.outdoorparking = outdoorparking;
        this.indoorparking = indoorparking;
        this.gardenfurniture = gardenfurniture;
        this.terrace = terrace;
        this.other = other;
        this.picture1 = picture1;
        this.picture2 = picture2;
        this.picture3 =picture3;
    }

    
    
    public House(int id, String country, String town, String nature, int price,
            String description, int rating, boolean airconditioner, boolean fireplace,
            boolean washingmachine, boolean dishwasher, boolean dvd, boolean indoorpool,
            boolean sauna, boolean jacuzzi, boolean tv, boolean elevator, boolean barbecue,
            boolean garden, boolean childrengames, boolean outdoorparking,
            boolean indoorparking, boolean gardenfurniture, boolean terrace, String other) {
        this.id = id;
        this.country = country;
        this.town = town;
        this.nature = nature;
        this.price = price;
        this.description = description;
        this.rating = rating;
        this.airconditioner = airconditioner;
        this.fireplace = fireplace;
        this.washingmachine = washingmachine;
        this.dishwasher = dishwasher;
        this.dvd = dvd;
        this.indoorpool = indoorpool;
        this.sauna = sauna;
        this.jacuzzi = jacuzzi;
        this.tv = tv;
        this.elevator = elevator;
        this.barbecue = barbecue;
        this.garden = garden;
        this.childrengames = childrengames;
        this.outdoorparking = outdoorparking;
        this.indoorparking = indoorparking;
        this.gardenfurniture = gardenfurniture;
        this.terrace = terrace;
        this.other = other;
        setEquipment();
    }

    
    public int getId() {
        return id;
    }

    public String getCountry() {
        return country;
    }

    public String getTown() {
        return town;
    }

    public String getNature() {
        return nature;
    }

    public int getPrice() {
        return price;
    }

    public String getDescription() {
        return description;
    }

    public int getRating() {
        return rating;
    }

    public boolean isAirconditioner() {
        return airconditioner;
    }

    public boolean isFireplace() {
        return fireplace;
    }

    public boolean isWashingmachine() {
        return washingmachine;
    }

    public boolean isDishwasher() {
        return dishwasher;
    }

    public boolean isDvd() {
        return dvd;
    }

    public boolean isIndoorpool() {
        return indoorpool;
    }

    public boolean isSauna() {
        return sauna;
    }

    public boolean isJacuzzi() {
        return jacuzzi;
    }

    public boolean isTv() {
        return tv;
    }

    public boolean isElevator() {
        return elevator;
    }

    public boolean isBarbecue() {
        return barbecue;
    }

    public boolean isGarden() {
        return garden;
    }

    public boolean isChildrengames() {
        return childrengames;
    }

    public boolean isOutdoorparking() {
        return outdoorparking;
    }

    public boolean isIndoorparking() {
        return indoorparking;
    }

    public boolean isGardenfurniture() {
        return gardenfurniture;
    }

    public boolean isTerrace() {
        return terrace;
    }

    public String getOther() {
        return other;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public void setTown(String town) {
        this.town = town;
    }

    public void setNature(String nature) {
        this.nature = nature;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setRating(int rating) {
        this.rating = rating;
    }

    public void setAirconditioner(boolean airconditioner) {
        this.airconditioner = airconditioner;
    }

    public void setFireplace(boolean fireplace) {
        this.fireplace = fireplace;
    }

    public void setWashingmachine(boolean washingmachine) {
        this.washingmachine = washingmachine;
    }

    public void setDishwasher(boolean dishwasher) {
        this.dishwasher = dishwasher;
    }

    public void setDvd(boolean dvd) {
        this.dvd = dvd;
    }

    public void setIndoorpool(boolean indoorpool) {
        this.indoorpool = indoorpool;
    }

    public void setSauna(boolean sauna) {
        this.sauna = sauna;
    }

    public void setJacuzzi(boolean jacuzzi) {
        this.jacuzzi = jacuzzi;
    }

    public void setTv(boolean tv) {
        this.tv = tv;
    }

    public void setElevator(boolean elevator) {
        this.elevator = elevator;
    }

    public void setBarbecue(boolean barbecue) {
        this.barbecue = barbecue;
    }

    public void setGarden(boolean garden) {
        this.garden = garden;
    }

    public void setChildrengames(boolean childrengames) {
        this.childrengames = childrengames;
    }

    public void setOutdoorparking(boolean outdoorparking) {
        this.outdoorparking = outdoorparking;
    }

    public void setIndoorparking(boolean indoorparking) {
        this.indoorparking = indoorparking;
    }

    public void setGardenfurniture(boolean gardenfurniture) {
        this.gardenfurniture = gardenfurniture;
    }

    public void setTerrace(boolean terrace) {
        this.terrace = terrace;
    }

    public void setOther(String other) {
        this.other = other;
    }

    @Override
    public int hashCode() {
        int hash = 17;
        
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final House other = (House) obj;
        if (this.id != other.id) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "House{" + "id=" + id + ", country=" + country + ", town=" + town + ", nature=" + nature + ", price=" + price + ", description=" + description + ", rating=" + rating + ", airconditioner=" + airconditioner + ", fireplace=" + fireplace + ", washingmachine=" + washingmachine + ", dishwasher=" + dishwasher + ", dvd=" + dvd + ", indoorpool=" + indoorpool + ", sauna=" + sauna + ", jacuzzi=" + jacuzzi + ", tv=" + tv + ", elevator=" + elevator + ", barbecue=" + barbecue + ", garden=" + garden + ", childrengames=" + childrengames + ", outdoorparking=" + outdoorparking + ", indoorparking=" + indoorparking + ", gardenfurniture=" + gardenfurniture + ", terrace=" + terrace + ", other=" + other + ", equipment=" + equipment + ", picture1=" + picture1 + ", picture2=" + picture2 + ", picture3=" + picture3 + ", rates=" + rates + ", comments=" + comments + '}';
    }

 
   
}
